code/datums/components/bakeable.dm

```diff
@@ -90,11 +90,18 @@
 	baked_result.pixel_y = original_object.pixel_y
 	used_tray.AddToPlate(baked_result)
 
+	var/list/asomnia_hadders = get_hearers_in_view(DEFAULT_MESSAGE_RANGE, used_oven) //need for asomnia work
+	for(var/mob/M in asomnia_hadders)
+		if(!HAS_TRAIT(M, TRAIT_ANOSMIA))
+			asomnia_hadders -= M
```

Comment:
Better var names. Also I feel like this should be reversed? 
```suggestion
	var/list/asomnia_hadders = list()
	for(var/mob/smeller in get_hearers_in_view(DEFAULT_MESSAGE_RANGE, used_oven))
		if(HAS_TRAIT(smeller, TRAIT_ANOSMIA))
			asomnia_hadders += smeller 
```

Reply:
Also it would be funny if these checks also excluded mobs with no head bodypart. No head = no nose?

Reply:
Sounds good... But then I need some more time to look at it. (Mobs without head?)